File: md5.go
package main

import (
	"encoding/json"
	"fmt"
)

type Input struct {
	Name string `json:"name"`
}

type Output struct {
	Result  int `json:"result"`
	Latency int `json:"latency"`
}

func hash(input string) int {
	hash := 0

	for i := 0; i < len(input); i++ {
		hash = (hash << 5) - hash + int(input[i])
		hash &= hash
	}

	return hash
}

func main(params json.RawMessage) (interface{}, error) {
	// Parse input parameters
	var input Input
	err := json.Unmarshal(params, &input)
	if err != nil {
		return nil, fmt.Errorf("failed to parse input: %v", err)
	}

	// If a value for name is provided, use it else use a default
	name := "stranger"
	if input.Name != "" {
		name = input.Name
	}

	output := Output{
		Result:  hash(name),
		Latency: 1,
	}

	return output, nil
}

File: md5.js
/**
 * @params is a JSON object with optional fields "name" and "place".
 * @return a JSON object containing the message in a field called "msg".
 */

function hash(input) {
  let hash = 0;

  for (let i = 0; i < input.length; i++) {
    hash = (hash << 5) - hash + input.charCodeAt(i);
    hash &= hash;
  }

  return hash;
}


function main(params) {
  // log the parameters to stdout


  // if a value for name is provided, use it else use a default
  var input = "stranger"

  output = hash(input)


  // construct the message using the values for name and place
  // return {msg:  'Hello, ' + name + ' from ' + place + '!'};
  return {"result":output,'latency':1}
}